### PR TITLE
Fix SPDX Tool bug in Scancode #2072

### PR DESCRIPTION
**Pull request by Jeeppler**
- Made SPDX Tool configurable
- Added checks to ensure the SPDX Tool is installed and called proberly
- Added script to generate test files

closes: #2072

### CODE REVIEW

File: sechub-pds-solutions/scancode/10-create-image.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ for <docker registry> with tag <version tag>.
 Required: <base image> ; for example ghcr.io/mercedes-benz/sechub/pds-base:v0.32.1
 
 Additionally these environment variables can be defined:
-- SCANCODE_VERSION - Scancode version to use. E.g. 31.2.1
+- SCANCODE_VERSION - Scancode version to use. E.g. 31.2.4
```
Comment (sven-dmlr) [inline sechub-pds-solutions/scancode/10-create-image.sh:19]: Is `SCANCODE_VERSION` mandatory or optional?
I think it's mandatory and should be checked in this script.

Please also document here `SPDX_TOOL_VERSION` and `SPDX_TOOL_CHECKSUM`
Where can I find out the version and the checksum and how is the expected formatting?

File: sechub-pds-solutions/scancode/10-create-image.sh [hunk 2]
```diff
@@ -49,6 +49,16 @@ if [[ ! -z "$SCANCODE_VERSION" ]] ; then
     BUILD_ARGS+=" --build-arg SCANCODE_VERSION=$SCANCODE_VERSION"
 fi
 
+if [[ ! -z "$SPDX_TOOL_VERSION" ]] ; then
```
Comment (sven-dmlr) [inline sechub-pds-solutions/scancode/10-create-image.sh:52]: If `SPDX_TOOL_VERSION` and `SPDX_TOOL_CHECKSUM` are unset, the Docker image build fails now.

Also blanks inside a Docker build argument (SPDX_TOOL_CHECKSUM) did not work for me:
```
> export SPDX_TOOL_VERSION="1.1.5"
> export SPDX_TOOL_CHECKSUM="feff2d5e7f66576a6e411d65a18046dbaf7ed6bad6e223a1c0ecee075a72b70b  tools-java-1.1.5-jar-with-dependencies.jar"
> export SCANCODE_VERSION=31.2.4
> ./10-create-image.sh my.registry/sechub/pds-scancode-test 0.999.0 my.registry/sechub/pds_base_image:latest
>> Base image: my.registry/sechub/pds_base_image:latest
>> Scancode version: 31.2.4
>> SPDX Tool version: 1.1.5
>> SPDX Tool checksum: feff2d5e7f66576a6e411d65a18046dbaf7ed6bad6e223a1c0ecee075a72b70b  tools-java-1.1.5-jar-with-dependencies.jar
"docker build" requires exactly 1 argument.
See 'docker build --help'.
...
```

As a solution, we could set `SPDX_TOOL_CHECKSUM="feff2d5e7f66576a6e411d65a18046dbaf7ed6bad6e223a1c0ecee075a72b70b"`
and the check in the Dockerfile (replace lines 52-54) like this:
```Dockerfile
echo "${SPDX_TOOL_CHECKSUM} tools-java-${SPDX_TOOL_VERSION}-jar-with-dependencies.jar" | sha256sum -c
```

**Review by sven-dmlr — CHANGES_REQUESTED** [review]
Looks good, but some things don't work.
Please check my comments/suggestions.

Also creating an extra checksums file in `Scancode-Debian.dockerfile` is not necessary.
You could replace lines 52-54 with:
```dockerfile
echo "$SPDX_TOOL_CHECKSUM" | sha256sum -c
```
(see also my comment in 10-create-image.sh regarding blanks in build args)

File: sechub-pds-solutions/scancode/docker/scripts/scancode.sh [hunk 3]
```diff
@@ -3,6 +3,23 @@
 
 source "${HELPER_FOLDER}/message.sh"
 
+function convert() {
+	echo "Converting to SPDX JSON"
+
+    cat "$spdx_file"
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-solutions/scancode/docker/scripts/scancode.sh:9]: Thanks, forgot to remove it.

**Review by sven-dmlr — APPROVED** [review]
It works now. Thank you.

A very good idea to download the checksum :+1:

**Comment by Jeeppler** [comment]
@sven-dmlr thanks for merging it.

**Pull request merged**